app/controllers/main/routes.py

```diff
@@ -58,14 +60,24 @@ def profilePage(username):
         rsvpedEventsList = EventRsvp.select().where(EventRsvp.user == profileUser)
         rsvpedEvents = [event.event.id for event in rsvpedEventsList]
 
+        allUserEntries = list(BackgroundCheck.select().where(BackgroundCheck.user == profileUser))
+        completedBackgroundCheck = {}
+        for entry in allUserEntries:
+            completedBackgroundCheck[entry.type.id] = entry.passBackgroundCheck
+        backgroundTypes = list(BackgroundCheckType.select())
+
+
         return render_template('/volunteer/volunteerProfile.html',
                                title="Volunteer Interest",
                                user = profileUser,
                                programs = programs,
                                interests = interests,
                                interests_ids = interests_ids,
                                upcomingEvents = upcomingEvents,
-                               rsvpedEvents = rsvpedEvents)
+                               rsvpedEvents = rsvpedEvents,
+                               backgroundTypes = backgroundTypes,
+                               completedBackgroundCheck = completedBackgroundCheck,
+                               currentUser = g.current_user)
```

Comment:
you don't need to pass a current user. the template can access g.current_user directly

Reply:
Fixed